src/script/sigcache.cpp

```diff
@@ -62,19 +62,11 @@ std::optional<std::pair<uint32_t, size_t>> CSignatureCache::setup_bytes(size_t n
     return setValid.setup_bytes(n);
 }
 
-/* In previous versions of this code, signatureCache was a local static variable
- * in CachingTransactionSignatureChecker::VerifySignature.  We initialize
- * signatureCache outside of VerifySignature to avoid the atomic operation per
- * call overhead associated with local static variables even though
- * signatureCache could be made local to VerifySignature.
-*/
-static CSignatureCache signatureCache;
-
 // To be called once in AppInitMain/BasicTestingSetup to initialize the
 // signatureCache.
-bool InitSignatureCache(size_t max_size_bytes)
+bool InitSignatureCache(size_t max_size_bytes, CSignatureCache& signature_cache)
```

Comment:
Any reason to not make `max_size_bytes` a `CSignatureCache` constructor param? That way `InitSignatureCache` could go away.